tools/convert-old-video-settings-file.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+"""Convert legacy video formats to generate gstreamer pipeline
+    Move old files, such as vidformat and gstreamer2 parameter files
+    to the user trash folder.
+"""
+
+import os
+import subprocess
+import sys
+
+
+def merge_gst_elements(elements: list):
+    """A simple helper to merge gstreamer elements in pipeline format
+
+    Args:
+        elements (list): List that contains gstreamer elements
+
+    Returns:
+        str: Final gstreamer pipeline
+    """
+
+    return ' ! '.join(elements)
+
+
+class Convert:
+    """Convert main class
+        Check file description for more information
+    """
+
+    HOME = os.path.expanduser('/tmp')
+    TRASH = os.path.join(HOME, '.local/share/Trash/files')
+    vidformat_file = os.path.join(HOME, 'vidformat.param')
+    gstreamer2_file = os.path.join(HOME, 'gstreamer2.param')
+
+    device_settings = {
+        'device': '/dev/video0',
+        'width': 1920,
+        'height': 1080,
+        'framerate': 30,
+    }
+
+    SOURCE_PIPELINE = merge_gst_elements([
+        'v4l2src device={device} do-timestamp=true',
+        'video/x-h264, width={width}, height={height}, framerate={framerate}/1',
+    ])
+
+    PARSE_PIPELINE = merge_gst_elements([
+        'h264parse',
+        'queue',
+        'rtph264pay config-interval=10 pt=96',
+        'udpsink host=192.168.2.1 port=5600',
+    ])
+
+    def update_configuration(self, pair: tuple):
+        """Update class internal configuration about video source and resolution
+
+        Args:
+            pair (tuple): Tuple that contains (key, value) for pipeline
+            settings dict
+        """
+
+        (key, value) = pair
+        self.device_settings[key] = value
+
+    @staticmethod
+    def create_source_element(device='/dev/video0', width=1920, height=1080, framerate=30):
```

Comment:
the h264 output is on /dev/video1. Will that be ok with this?

Reply:
This is just the default parameters, if the user has any other configuration it'll be used what the user has in his gst/video parameter files.

Reply:
oh i see